src/callbacks_step/amr.jl

```diff
@@ -346,6 +356,168 @@ function (amr_callback::AMRCallback)(u_ode::AbstractVector, mesh::TreeMesh,
     return has_changed
 end
 
+# `passive_args` is currently used for Euler with self-gravity to adapt the gravity solver
+# passively without querying its indicator, based on the assumption that both solvers use
+# the same mesh. That's a hack and should be improved in the future once we have more examples
+# and a better understanding of such a coupling.
+# `passive_args` is expected to be an iterable of `Tuple`s of the form
+# `(p_u_ode, p_mesh, p_equations, p_dg, p_cache)`.
+function (amr_callback::AMRCallback)(u_ode::AbstractVector, mesh::TreeMesh,
+                                     equations, dg::DG,
+                                     cache, cache_parabolic,
+                                     semi::SemidiscretizationHyperbolicParabolic,
+                                     t, iter;
+                                     only_refine = false, only_coarsen = false,
+                                     passive_args = ())
+    @unpack controller, adaptor = amr_callback
+
+    u = wrap_array(u_ode, mesh, equations, dg, cache)
+    # TODO: Keep indicator based on hyperbolic variables?
```

Comment:
Indicators are still based only on the hyperbolic variables.

Reply:
I think for now this is ok to keep.you can thus just remove the TODO note here and leave the informational comment, then resolve this convo.